File: studio/AuraStudio/Tests/AuraStudioTests/MediaCategoryHeuristicsTests.swift
import XCTest
@testable import AuraStudio

/// D-192: heuristicas puras de categorizacion de fotos/video -- sin
/// tocar disco, la lectura real de EXIF/duracion vive en
/// `MediaCategoryClassifier` (no testeado aca, necesita archivos reales).
final class MediaCategoryHeuristicsTests: XCTestCase {
    // MARK: - Fotos

    func testPhotoWithKnownAISoftwareTagIsAIGenerated() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyPhoto(softwareTag: "Midjourney v6", hasCameraExif: false), .aiGenerated)
        XCTAssertEqual(MediaCategoryHeuristics.classifyPhoto(softwareTag: "Adobe Firefly", hasCameraExif: true), .aiGenerated)
    }

    func testPhotoWithCameraExifAndNoAITagIsPhoto() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyPhoto(softwareTag: nil, hasCameraExif: true), .photos)
    }

    func testPhotoWithNoCameraExifAndNoAITagIsImage() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyPhoto(softwareTag: nil, hasCameraExif: false), .images)
    }

    func testPhotoSoftwareTagMatchIsCaseInsensitive() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyPhoto(softwareTag: "STABLE DIFFUSION XL", hasCameraExif: false), .aiGenerated)
    }

    func testUnrelatedSoftwareTagDoesNotTriggerAIGenerated() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyPhoto(softwareTag: "Adobe Photoshop 25.0", hasCameraExif: true), .photos)
    }

    // MARK: - Videos

    func testShortVideoIsHomeVideo() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyVideo(durationSeconds: 45), .homeVideos)
        XCTAssertEqual(MediaCategoryHeuristics.classifyVideo(durationSeconds: 180), .homeVideos)
    }

    func testMediumVideoIsVideos() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyVideo(durationSeconds: 600), .videos)
    }

    func testLongVideoIsMovie() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyVideo(durationSeconds: 5400), .movies)
    }

    func testUnknownDurationFallsBackToVideos() {
        XCTAssertEqual(MediaCategoryHeuristics.classifyVideo(durationSeconds: nil), .videos)
        XCTAssertEqual(MediaCategoryHeuristics.classifyVideo(durationSeconds: 0), .videos)
    }
}

File: studio/windows/tests/AuraStudio.Core.Tests/MediaCategoryHeuristicsTests.cs
using AuraStudio.Core;
using Xunit;

namespace AuraStudio.Core.Tests;

/// <summary>
/// D-192/D-228: heurísticas puras de categorización de fotos/video — sin tocar
/// disco, la lectura real de EXIF/duración vive en `MediaCategoryClassifier`
/// (no testeado acá, necesita archivos reales).
/// </summary>
public class MediaCategoryHeuristicsTests
{
    // MARK: - Fotos

    [Fact]
    public void PhotoWithKnownAISoftwareTagIsAIGenerated()
    {
        // Equivalente a testPhotoWithKnownAISoftwareTagIsAIGenerated
        Assert.Equal("IA", MediaCategoryHeuristics.ClassifyPhoto("Midjourney v6", hasCameraExif: false));
        Assert.Equal("IA", MediaCategoryHeuristics.ClassifyPhoto("Adobe Firefly", hasCameraExif: true));
    }

    [Fact]
    public void PhotoWithCameraExifAndNoAITagIsPhoto()
    {
        // Equivalente a testPhotoWithCameraExifAndNoAITagIsPhoto
        Assert.Equal("Fotos", MediaCategoryHeuristics.ClassifyPhoto(null, hasCameraExif: true));
    }

    [Fact]
    public void PhotoWithNoCameraExifAndNoAITagIsImage()
    {
        // Equivalente a testPhotoWithNoCameraExifAndNoAITagIsImage
        Assert.Equal("Imágenes", MediaCategoryHeuristics.ClassifyPhoto(null, hasCameraExif: false));
    }

    [Fact]
    public void PhotoSoftwareTagMatchIsCaseInsensitive()
    {
        // Equivalente a testPhotoSoftwareTagMatchIsCaseInsensitive
        Assert.Equal("IA", MediaCategoryHeuristics.ClassifyPhoto("STABLE DIFFUSION XL", hasCameraExif: false));
    }

    [Fact]
    public void UnrelatedSoftwareTagDoesNotTriggerAIGenerated()
    {
        // Equivalente a testUnrelatedSoftwareTagDoesNotTriggerAIGenerated
        Assert.Equal("Fotos", MediaCategoryHeuristics.ClassifyPhoto("Adobe Photoshop 25.0", hasCameraExif: true));
    }

    // MARK: - Videos

    // D-228: se eliminó el corte de "casero" (<= 3 min) — ya no hay heurística
    // automática para "Series", el usuario la asigna a mano. Un video corto ahora
    // cae en el default (.videos), igual que cualquier duración que no sea
    // claramente una película.
    [Theory]
    [InlineData(45)]
    [InlineData(180)]
    public void ShortVideoIsVideos(double durationSeconds)
    {
        // Equivalente a testShortVideoIsVideos
        Assert.Equal(MediaCategory.Videos, MediaCategoryHeuristics.ClassifyVideo(durationSeconds));
    }

    [Fact]
    public void MediumVideoIsVideos()
    {
        // Equivalente a testMediumVideoIsVideos
        Assert.Equal(MediaCategory.Videos, MediaCategoryHeuristics.ClassifyVideo(600));
    }

    [Fact]
    public void LongVideoIsMovie()
    {
        // Equivalente a testLongVideoIsMovie — el corte es > 2400 s (40 min)
        Assert.Equal(MediaCategory.Movies, MediaCategoryHeuristics.ClassifyVideo(5400));
    }

    [Fact]
    public void UnknownDurationFallsBackToVideos()
    {
        // Equivalente a testUnknownDurationFallsBackToVideos
        Assert.Equal(MediaCategory.Videos, MediaCategoryHeuristics.ClassifyVideo(null));
        Assert.Equal(MediaCategory.Videos, MediaCategoryHeuristics.ClassifyVideo(0));
    }
}
